docs/queries/grouping.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: default
+title: Grouping
+nav_order: 3
+parent: Queries
+---
+
+# Grouping
+{: .no_toc }
+
+<details open markdown="block">
+  <summary>
+    Table of contents
+  </summary>
+  {: .text-delta }
+1. TOC
+{:toc}
+</details>
+
+---
+
+## Basics
+
+By default Tasks displays tasks in a single list.
+
+To divide the matching tasks up with headings, you can use the `group by` line to the query.
+
+You can group by the following properties:
+
+File locations:
+
+1. `path` (the path to the file that contains the task, that is, the folder and the filename)
+2. `folder` (the folder to the file that contains the task, which will be `/` for files in root of the vault)
+3. `filename` (the filename of the file that contains the task, without the `.md` extension)
+
+File contents:
+
+1. `linktext` (the text that would be shown in the task's backlink, but with no link added)
+2. `heading` (the heading preceding the task, or `(No heading)` if there are no headings in the file)
+
+Task properties:
+
+1. `status` (Done or Todo, which is capitalized for visibility in the headings)
+
+You can add multiple `group by` query options, each on an extra line.
+This will create nested groups.
+The first group has the highest priority.
+
+Each subsequent `group by` will generate a new heading-level within the existing grouping:
+
+- First `group by` is displayed as `h4` headings
+- Second `group by` is displayed as `h5` headings
+- Third and subsequent `group by` are displayed as `h6` headings
+
+---
+
+## Screenshots
+
+**TODO Change screenshot URLs to refer to main repo and branch**
+
+Here is an example Tasks result, without any `group by` commands:
+
+![Tasks Ungrouped](https://github.com/claremacrae/obsidian-tasks/raw/add-group-by-feature/resources/screenshots/tasks_ungrouped.png)
```

Comment:
Relative links do not work?

Reply:
I don’t know, so I stuck with being consistent with the other uses of images in the docs.

Reply:
Relative links don't work in the main README, as the document will be shown within obsidian when looking at the plugin.

In this case, you link to a branch of your fork which you are likely to delete at some point. Is that on purpose?

Reply:
I just now read your todo 😬

Reply:
> Relative links don't work in the main README, as the document will be shown within obsidian when looking at the plugin.

I think what you are probably saying is that I actually could use absolute links now. I will do this...

(And thanks for asking, @sytone!)

> In this case, you link to a branch of your fork which you are likely to delete at some point. Is that on purpose?

Yes - see the TODO... I wanted people to be able to see the images whilst the PR is being worked on, and assumed that absolute links were needed...

```text
+**TODO Change screenshot URLs to refer to main repo and branch**
+
+Here is an example Tasks result, without any `group by` commands:
```

Reply:
I changed the image links to be relative (don't worry about the `//`):
https://github.com/claremacrae/obsidian-tasks/commit/657be7ed4d8781d13cfaee62017e3328eb870cf9

The images then work fine in WebStorm still.

I've just seen that they give this via Jekyll:

![image](https://user-images.githubusercontent.com/4840096/167263459-76d11488-2b82-4f37-bada-caa6e25e3160.png)

(Yay for testing the docs locally!)

I tried these relative paths, and I couldn't get any of them to work:

```text
![Tasks Ungrouped](/assets/images/tasks_ungrouped.png)
Tasks not grouped.

![Tasks Ungrouped](../assets/images/tasks_ungrouped.png)
Tasks not grouped.

![Tasks Ungrouped](../../assets/images/tasks_ungrouped.png)
Tasks not grouped.
```

.... before eventually realising that the files in the `resources/screenshots` folder are not copied in to the `_site` folder, so won't currently be available on on GitHub Pages to be viewed from inside the server files.

I think for now I'll go back to user the nasty hard-coded URLs to my fork, for testing, and change them to the main server just before finalising the PR.

We can fix the publishing of screenshots later, if anyone thinks it's worth doing...

Reply:
I have changed the URLs now, to refer to images on the main project, main branch.